MsixCore/msixmgr/msixmgr_tracelogging.h

```diff
@@ -0,0 +1,168 @@
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA));
+    }
+
+    inline void TraceLogApplyACLsWorkflow(const wchar_t* WorkflowID, const wchar_t* PackageName)
+    {
+        TraceLoggingWrite(
+            g_MsixMgrTelemetryProvider,
+            "ApplyACLsWorkflow",
+            TraceLoggingWideString(WorkflowID, "WorkflowID"),
+            TraceLoggingWideString(PackageName, "PackageName"),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA));
+    }
+
+    inline void TraceLogMountWorkflow(const wchar_t* WorkflowID, const wchar_t* MountImageType)
+    {
+        TraceLoggingWrite(
+            g_MsixMgrTelemetryProvider,
+            "MountWorkflow",
+            TraceLoggingWideString(WorkflowID, "WorkflowID"),
+            TraceLoggingWideString(MountImageType, "MountImageType"),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA));
+    }
+
+    inline void TraceLogUnmountWorkflow(const wchar_t* WorkflowID, const wchar_t* UnmountImageType)
+    {
+        TraceLoggingWrite(
+            g_MsixMgrTelemetryProvider,
+            "UnmountWorkflow",
+            TraceLoggingWideString(WorkflowID, "WorkflowID"),
+            TraceLoggingWideString(UnmountImageType, "UnmountImageType"),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA));
+    }
+
+    inline std::string GuidToString(GUID* guid) {
+        char guid_string[37];
+        snprintf(
+            guid_string, sizeof(guid_string),
+            "%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x",
+            guid->Data1, guid->Data2, guid->Data3,
+            guid->Data4[0], guid->Data4[1], guid->Data4[2],
+            guid->Data4[3], guid->Data4[4], guid->Data4[5],
+            guid->Data4[6], guid->Data4[7]);
+        return guid_string;
+    }
+
+    inline std::wstring CreateWorkflowId() {
+        GUID guid;
+        HRESULT hr = CoCreateGuid(&guid);
+
+        std::wstring WorkflowId = utf8_to_utf16(GuidToString(&guid));
+
+        return WorkflowId;
+    }
+
+    inline double CalcWorkflowElapsedTime(LARGE_INTEGER MsixMgrLoad_StartCounter,
+        LARGE_INTEGER MsixMgrLoad_EndCounter, LARGE_INTEGER MsixMgrLoad_Frequency)
+    {
+        double WorkflowElapsedTime = (MsixMgrLoad_EndCounter.QuadPart - MsixMgrLoad_StartCounter.QuadPart) / (double)MsixMgrLoad_Frequency.QuadPart;
+
+        return WorkflowElapsedTime;
+    }
+
+    inline std::wstring GetErrorCodeFromHRESULT(HRESULT hr)
+    {
+        std::stringstream ErrorCodeStream;
+        ErrorCodeStream << "0x" << std::hex << hr;
+        std::wstring ErrorCode = utf8_to_utf16(ErrorCodeStream.str());
+
+        return ErrorCode;
+    }
+
+    inline std::wstring ExtractPackageNameFromFilePath(std::wstring FilePath)
+    {
+        size_t pos = FilePath.find_last_of('\\');
```

Comment:
Are we sure packagePath will always have "\\" inside it?
Will there be cases where path can be ".\vlc.msix" or "C:\vlc.msix"?

Reply:
As discussed offline, this code change is fine and the comment can be closed